tools/runners/test262-harness.py

```diff
@@ -0,0 +1,913 @@
+            elif 'onlyStrict' in flags:
+                raise TypeError(
+                    "The `raw` flag is incompatible with the `onlyStrict` flag")
+            elif self.get_include_list():
+                raise TypeError(
+                    "The `raw` flag is incompatible with the `includes` tag")
+
+
+class ProgressIndicator(object):
+
+    def __init__(self, count):
+        self.count = count
+        self.succeeded = 0
+        self.failed = 0
+        self.failed_tests = []
+
+    def has_run(self, result):
+        result.report_outcome(True)
+        if result.has_unexpected_outcome():
+            self.failed += 1
+            self.failed_tests.append(result)
+        else:
+            self.succeeded += 1
+
+
+def make_plural(num):
+    if num == 1:
+        return (num, "")
+    return (num, "s")
+
+
+def percent_format(partial, total):
+    return "%i test%s (%.1f%%)" % (make_plural(partial) +
+                                   ((100.0 * partial)/total,))
+
+
+class TestSuite(object):
+
+    def __init__(self, root, strict_only, non_strict_only, unmarked_default, print_handle, exclude_list_path):
+        self.test_root = path.join(root, 'test')
+        self.lib_root = path.join(root, 'harness')
+        self.strict_only = strict_only
+        self.non_strict_only = non_strict_only
+        self.unmarked_default = unmarked_default
+        self.print_handle = print_handle
+        self.include_cache = {}
+        self.exclude_list = []
+        self.logf = None
+
+        if exclude_list_path:
+            if os.path.exists(exclude_list_path):
+                self.exclude_list = xml.dom.minidom.parse(exclude_list_path)
+                self.exclude_list = self.exclude_list.getElementsByTagName("test")
+                self.exclude_list = [x.getAttribute("id") for x in self.exclude_list]
+            else:
+                report_error("Couldn't find excludelist '%s'" % exclude_list_path)
+
+    def validate(self):
+        if not path.exists(self.test_root):
+            report_error("No test repository found")
+        if not path.exists(self.lib_root):
+            report_error("No test library found")
+
+    @staticmethod
+    def is_hidden(test_path):
+        return test_path.startswith('.') or test_path == 'CVS'
+
+    @staticmethod
+    def is_test_case(test_path):
+        return test_path.endswith('.js') and not test_path.endswith('_FIXTURE.js')
+
+    @staticmethod
+    def should_run(rel_path, tests):
+        if not tests:
+            return True
+        for test in tests:
+            if test in rel_path:
+                return True
+        return False
+
+    def get_include(self, name):
+        if not name in self.include_cache:
+            static = path.join(self.lib_root, name)
+            if path.exists(static):
+                file_desc = open(static)
+                contents = file_desc.read()
+                contents = re.sub(r'\r\n', '\n', contents)
```

Comment:
Please use the following style here:
```python
with open(...) as fp:
  contents = fp.read()
contents = re.sub(...)
```

then the `file_desc.close` can be removed from below.